Bottom: 370d2e9d02288bf5f125beaf02726dc246efc1c4
Top:    9a4a62432469edb86ba0fef4e959893f7b116e38
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-01-11 14:11:38 +0800

Refresh of vhost-prefetch-next-descriptor

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 5849b37..116b5a2 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -28,6 +28,7 @@
 #include <linux/module.h>
 #include <linux/sort.h>
 #include <linux/interval_tree_generic.h>
+#include <linux/prefetch.h>
 
 #include "vhost.h"
 
@@ -1813,6 +1814,8 @@ static unsigned next_desc(struct vhost_virtqueue *vq, struct vring_desc *desc)
 	 * architectures only need a compiler barrier here. */
 	read_barrier_depends();
 
+	prefetchw(vq->desc + next);
+
 	return next;
 }
